pkg/driver/aws-ebs/aws_ebs_tags_controller.go

```diff
@@ -182,33 +190,49 @@ func (c *EBSVolumeTagsController) createSessionWithCredentials(credentialsData [
 	}
 
 	// Create base AWS session
-	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String(region),
-	})
+	// sess, err := session.NewSession(&aws.Config{
+	// 	Region: aws.String(region),
+	// })
+	// if err != nil {
+	// 	klog.Errorf("Error creating base AWS session: %v", err)
+	// 	return nil, fmt.Errorf("error creating AWS session: %v", err)
+	// }
+
+	awsConfig, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(region))
```

Comment:
Eventually, we will need to fix those `.TODO()` contexts. I did not check all of them but e.g. this one is easy - `getEC2Client` already has a context, it must pass it through `createAWSSession` -> `createSessionWithCredentials`.